lib/administration/user/src/user_administration.cpp

```diff
@@ -85,18 +85,29 @@ namespace irods::experimental::administration::NAMESPACE_IMPL
 
     auto add_group(RxComm& _comm, const group& _group) -> void
     {
-        const auto zone = get_local_zone(_comm);
+        const auto zone{get_local_zone(_comm)};
 
-        GeneralAdminInput input{};
-        input.arg0 = "add";
-        input.arg1 = "user";
-        input.arg2 = _group.name.data();
-        input.arg3 = "rodsgroup";
-        input.arg4 = zone.data();
+        const auto current_user{user{_comm.clientUser.userName, _comm.clientUser.rodsZone}};
+        const auto user_type{type(_comm, current_user)};
 
-        if (const auto ec = rxGeneralAdmin(&_comm, &input); ec != 0) {
-            // NOLINTNEXTLINE(cppcoreguidelines-pro-bounds-array-to-pointer-decay)
-            THROW(ec, fmt::format("Failed to add group [{}].", _group.name));
+        if (user_type == irods::experimental::administration::user_type::groupadmin) {
+            userAdminInp_t input{.arg0 = "mkgroup",
+                                 .arg1 = const_cast<char*>(_group.name.data()),
+                                 .arg2 = "rodsgroup",
+                                 .arg3 = const_cast<char*>(zone.data())};
```

Comment:
Are you `const_cast`ing away `const` here? Are you guaranteed that none of these pointers are ever modified downstream? Could be UB otherwise.

Reply:
Going to close this for now, unless someone else has issues with the const_cast.